test/registry.js

```diff
@@ -240,3 +240,65 @@ t.test('packument that falls back to fullMetadata', t => {
     t.end()
   })
 })
+
+t.test('option replaceRegistryHost', rhTest => {
+  const nock = require('nock')
+  const { join, resolve } = require('path')
+  const fs = require('fs')
+
+  const abbrev = resolve(__dirname, 'fixtures/abbrev-1.1.1.tgz')
+  const abbrevTGZ = fs.readFileSync(abbrev)
+  const testdir = t.testdir()
+
+  nock('https://registry.github.com')
+    .persist()
+    .get('/abbrev')
+    .reply(200, JSON.stringify({
+      _id: 'abbrev',
+      _rev: 'lkjadflkjasdf',
+      name: 'abbrev',
+      'dist-tags': { latest: '1.1.1' },
+      versions: {
+        '1.1.1': {
+          name: 'abbrev',
+          version: '1.1.1',
+          dist: {
+            tarball: 'https://registry.npmjs.org/abbrev/-/abbrev-1.1.1.tgz',
+          },
+        },
+      },
+    }))
+
+  nock('https://registry.npmjs.org')
```

Comment:
check out `test/fixtures/tnock.js` in the cli to see a pattern we're landing on for using nock with tap